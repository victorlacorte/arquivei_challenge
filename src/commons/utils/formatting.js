function toBRL(number) {
  return number.toLocaleString('pt-BR', { style: 'currency', currency: 'BRL' });
}

export {
  toBRL,
};
