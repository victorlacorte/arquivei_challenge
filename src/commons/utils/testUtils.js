/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from 'react';
import { ThemeProvider } from 'styled-components';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mount, shallow } from 'enzyme';

import defaultTheme from 'commons/styles/theme';

function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

function shallowWithTheme(child) {
  return shallow(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

export {
  mountWithTheme,
  shallowWithTheme,
};
