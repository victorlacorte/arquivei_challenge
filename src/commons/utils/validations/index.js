export { default as cnpjValidations } from './cnpj';
export { default as consultKeysValidations } from './consultKeys';
export { default as creditCardValidations } from './creditCard';
export { default as formValidations } from './formValidations';
