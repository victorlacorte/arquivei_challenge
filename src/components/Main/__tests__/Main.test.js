import React from 'react';

import Main from 'components/Main';
import { shallowWithTheme } from 'commons/utils/testUtils';

describe('<Main />', () => {
  it('creates a Main component', () => {
    const wrapper = shallowWithTheme(<Main />);

    expect(wrapper).toMatchSnapshot();
  });
});
