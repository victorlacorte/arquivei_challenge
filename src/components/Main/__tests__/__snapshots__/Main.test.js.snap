// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> creates a Main component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "styles__StyledMain-sc-1sa92va-0",
        "isStatic": false,
        "rules": Array [
          "box-sizing:border-box;position:relative;margin:0 auto;min-height:calc(100vh - 92px - 60px);min-width:320px;overflow:hidden;padding:24px;",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
        ],
      },
      "displayName": "styles__StyledMain",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "styles__StyledMain-sc-1sa92va-0",
      "target": "main",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
